compiler-plugin/src/main/java/io/ballerina/stdlib/tcp/compiler/TcpServiceValidator.java

```diff
@@ -63,12 +64,14 @@ public void validate() {
                     ExpressionNode expressionNode = returnStatementNode.expression().get();
 
                     if (expressionNode instanceof ExplicitNewExpressionNode) { // handle return new HelloService();
-                        TypeReferenceTypeSymbol typeReferenceTypeSymbol = (TypeReferenceTypeSymbol) ctx.semanticModel()
-                                .type(expressionNode).get();
-                        ClassSymbol classSymbol = (ClassSymbol) typeReferenceTypeSymbol.typeDescriptor();
-                        TcpConnectionServiceValidator tcpConnectionServiceValidator =
-                                new TcpConnectionServiceValidator(ctx, classSymbol);
-                        tcpConnectionServiceValidator.validate();
+                        TypeSymbol symbol = ctx.semanticModel().type(expressionNode).get();
+                        if (symbol instanceof TypeReferenceTypeSymbol) {
+                            TypeReferenceTypeSymbol typeReferenceTypeSymbol = (TypeReferenceTypeSymbol) symbol;
+                            ClassSymbol classSymbol = (ClassSymbol) typeReferenceTypeSymbol.typeDescriptor();
+                            TcpConnectionServiceValidator tcpConnectionServiceValidator =
+                                    new TcpConnectionServiceValidator(ctx, classSymbol);
+                            tcpConnectionServiceValidator.validate();
+                        }
```

Comment:
I think better to have an validation to check whether symbol is present or not since this could also lead to a plugin-crash. We could do a negative check and continue the loop and it will reduce line width. This would improve the readability as well.
```suggestion
                        Optional<TypeSymbol> symbolOpt = ctx.semanticModel().type(expressionNode);
                        if (symbolOpt.isEmpty()) {
                            continue;
                        }
                        TypeSymbol symbol = symbolOpt.get();
                        if (!(symbol instanceof TypeReferenceTypeSymbol)) {
                            continue;
                        }
                        
                        TypeReferenceTypeSymbol typeReferenceTypeSymbol = (TypeReferenceTypeSymbol) symbol;
                        ClassSymbol classSymbol = (ClassSymbol) typeReferenceTypeSymbol.typeDescriptor();
                        TcpConnectionServiceValidator tcpConnectionServiceValidator =
                                    new TcpConnectionServiceValidator(ctx, classSymbol);
                        tcpConnectionServiceValidator.validate();
```